### DISCUSSION TITLE
[Feature] Generate thumbnails only once for each media

### DISCUSSION THREAD

**Discussion by dronnikovigor** [immich-app/immich#11231] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Now, if I have several users with personal libraries, that utilize the same media path (so, basically different users see the same media), thumbnails are generated for each user separately. So, one photo may have 1,2,3 or more copies of thumbnails. This is not efficient use of disk space. 
What I suggest is to generate thumbnails only once for photo across all users (based on md5 e.g.).

P.S. It's not an issue, if library is shared from one user to another. It's about the case when different users have personal library with same media path.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by mertalev** [comment]
I think this suggestion makes sense. My only concern is that a user can delete a library asset, which means the thumbnail is deleted for everyone else too. But the main asset would be deleted for everyone as well in this case, so this issue already exists.

**Reply by dronnikovigor** [reply]
Yeah, it seems like now if one user deletes media from library, that uses "common" path, another users also will lose this media. So, same thing with thumbnails, I guess this shouldn't be a blocker

**Comment by bo0tzz** [comment]
#4542
